tests/test_dose.py

```diff
@@ -0,0 +1,344 @@
+"""unittest cases for dose."""
+# test_dvhdose.py
+# Copyright (c) 2016-2020 Aditya Panchal
+# Copyright (c) 2020 Dan Cutright
+
+
+from __future__ import division
+import unittest
+import os
+from dicompylercore import dicomparser, dose
+
+try:
+    from pydicom.dataset import Dataset
+    from pydicom.sequence import Sequence
+    from pydicom import read_file as read_dicom
+except ImportError:
+    from dicom.dataset import Dataset
+    from dicom.sequence import Sequence
+    from dicom import read_file as read_dicom
+from numpy import arange, zeros
+from numpy.testing import (
+    assert_array_almost_equal,
+    assert_array_equal,
+    assert_raises,
+)
+import warnings
+
+basedata_dir = "tests/testdata"
+example_data = os.path.join(basedata_dir, "example_data")
+
+
+def assert_array_not_equal(arr_1, arr_2):
+    return assert_raises(AssertionError, assert_array_equal, arr_1, arr_2)
+
+
+def assert_array_not_almost_equal(arr_1, arr_2):
+    return assert_raises(
+        AssertionError, assert_array_almost_equal, arr_1, arr_2
+    )
+
+
+class TestDose(unittest.TestCase):
+    """Unit tests for Dose calculations."""
+
+    def setUp(self):
+        """Setup files for common case testing."""
+        self.rtdose_dcm = os.path.join(example_data, "rtdose.dcm")
+        self.rtdose = dicomparser.DicomParser(self.rtdose_dcm)
+
+        self.dvhs = self.rtdose.GetDVHs()
+
+        self.dosegrid = dose.DoseGrid(self.rtdose_dcm)
+
+    def test_modality_check(self):
+        """Test non-RTDOSE raises AttributeError"""
+        ds = dicomparser.DicomParser(self.rtdose_dcm).ds
+        ds.Modality = "RTPLAN"
+        with self.assertRaises(AttributeError):
+            dose.DoseGrid(ds)
+
+    def test_shape(self):
+        """Test dose grid shape extraction from the DICOM data."""
+        shape = self.dosegrid.shape
+
+        # x-dimension
+        self.assertEqual(shape[0], 194)
+        # y-dimension
+        self.assertEqual(shape[1], 129)
+        # z-dimension
+        self.assertEqual(shape[2], 98)
+
+    def test_scale(self):
+        """Test dose grid resolution extraction from the DICOM data."""
+        scale = self.dosegrid.scale
+
+        # x-dimension
+        self.assertEqual(scale[0], 2.5)
+        # y-dimension
+        self.assertEqual(scale[1], 2.5)
+        # z-dimension
+        self.assertEqual(scale[2], 3.0)
+
+    def test_offset(self):
+        """Test dose grid resolution extraction from the DICOM data."""
+        offset = self.dosegrid.offset
+
+        # x-dimension
+        self.assertEqual(offset[0], -228.6541915)
+        # y-dimension
+        self.assertEqual(offset[1], -419.2444776)
+        # z-dimension
+        self.assertEqual(offset[2], -122.4407)
+
+    def test_is_coincident(self):
+        """Test spatial coincidence of two dose grids"""
+
+        # Self coincidence
+        other = dose.DoseGrid(self.rtdose_dcm)
+        self.assertEqual(self.dosegrid.is_coincident(other), True)
+
```

Comment:
For these lines comparing to bool values, should be able to use `self.assertTrue`  or `self.assertFalse` which I think is easier to understand on a quick read.

E.g.
```python
self.assertTrue(self.dosegrid.is_coincident(other))
```

Reply:
I like it.  This is my first unit test, so these are good things to learn.